Lib/fontmake/instantiator.py

```diff
@@ -224,21 +226,33 @@ def from_designspace(
         # because the math behind varLib and MutatorMath uses the default font as the
         # point of reference for all data.
         default_font = designspace.default.font
+        non_default_layer_name = designspace.default.layerName
+        non_default_layer_flag = (
+            non_default_layer_name is not None
+            and non_default_layer_name != DEFAULT_LAYER_NAME
+        )
+
         glyph_names: Set[str] = set(default_font.keys())
 
-        for source in designspace.sources:
-            other_names = set(source.font.keys())
-            diff_names = other_names - glyph_names
-            if diff_names:
-                logger.warning(
-                    "The source %s (%s) contains glyphs that are missing from the "
-                    "default source, which will be ignored: %s. If this is unintended, "
-                    "check that these glyphs have the exact same name as the "
-                    "corresponding glyphs in the default source.",
-                    source.name,
-                    source.filename,
-                    ", ".join(sorted(diff_names)),
-                )
+        if non_default_layer_flag:
+            if non_default_layer_name in default_font.layers:
```

Comment:
you are not handling the `else:` here. We don't want to silently pass if the specified surce.layerName actually is not in the font. We want to explode! 
Do:

```python
try:
    layer = default_font.layers[non_default_layer_name]
except KeyError as e:
    raise InstantiatorError(f"Layer {non_default_layer_name!r} not found in {source.filename}") from e
```

Reply:
OK fixed.